utils/httpUtils.ts

```diff
@@ -91,3 +147,40 @@ export const notFound = (message: string) => {
     message: message,
   });
 };
+
+export const getFeatureFlags = async (context: any) => {
+  try {
+    const response = await fetch(process.env.API_URL + "/users/feature_flags", {
+      method: "GET",
+      headers: {
+        Cookie: context.request.headers.get("cookie"),
+        "Content-Type": "application/json",
+      },
+    });
+    return await response.json();
+  } catch (e) {
+    return Promise.reject(e.response);
+  }
+};
+
+export const getFeatureFlagsFromRequest = (context) => {
+  return context.request.headers.get("readFromNextGen");
+};
+
+export const isNextGenEnabled = async (context) => {
+  let readFromNextGen = getFeatureFlagsFromRequest(context);
+  if (readFromNextGen !== null) {
+    // if the header is set, return the value
+    return readFromNextGen;
+  }
+  try {
+    const featureFlags = await getFeatureFlags(context);
+    const combinedFeatureFlags = featureFlags["launched_feature_list"].concat(
```

Comment:
Question - why are there 2 places where the `next_gen` flag could be?

Reply:
I thought we are planning to only get the feature flag via the headers, and not querying them from Rails.  Do we need that functionality at all?

Reply:
I just saw @saq7 's previous comment about this: https://github.com/chanzuckerberg/czid-graphql-federation-server/pull/44#issuecomment-1889595336

I'm skeptical we should add this unless we had a particular use in mind.  We can save off the code for fetching from Rails in a branch or wiki if we'd like, but I'd recommend we just stick to the header pattern and require clients to make requests that way, until we encounter a use case where that won't work.

Reply:
That is how the rails endpoint is set up. `launched_feature_list` is a universal list for all users and `allowed_feature_list` is user specific

Reply:
I commented out that code for the merge. I suggest we leave the comments in place so that they can be enabled for local development until we are in the flow of building using this pattern